### PR TITLE
Browser Detection

### PR DESCRIPTION
**Pull request by mathewleblanc**
# Browser Detection

In response to issue #520 I have added browser detection to the upload page. The user's first visit to the page is saved as a boolean in session storage. If they are using an unsupported browser (anything other than Chrome and Firefox) an alert appears with an error message informing them that their current browser is not supported. This alert will not appear again when the user visits the page until the current tab is closed and they visit again or they clear their cache/history. This should be a decent temporary fix until we decide to use Modernizr as Scott suggested.

### CODE REVIEW

File: lexos/static/js/scripts_upload.js [hunk 1]
```diff
@@ -194,4 +194,51 @@ $(function () {
...
     Init()
   }
+
+  browserDetection()
 })
+
+function browserDetection () {
+  // Display a warning if using an unsupported browser
+
+  var browserWarningMsg = `You are currently using an unsupported browser.\
```
Comment (Weiqi97) [inline lexos/static/js/scripts_upload.js]: Please use ```const``` if this variable does not get changed. Same for the couple declarations below.

File: lexos/static/js/scripts_upload.js [hunk 2]
```diff
@@ -194,4 +194,51 @@ $(function () {
   if (window.File && window.FileList && window.FileReader) {
     Init()
   }
+
+  browserDetection()
 })
+
+function browserDetection () {
```
Comment (Weiqi97) [inline lexos/static/js/scripts_upload.js]: Please add a documentation here for this function.

File: lexos/static/js/scripts_upload.js [hunk 3]
```diff
@@ -194,4 +194,51 @@ $(function () {
...
+
+  // Chrome 1+
+  var isChrome = !!window.chrome && !!window.chrome.webstore;
+
+  // If first time on page, show error message if browser is not supported
+  if (typeof(Storage) !== "undefined") {
+    if (!sessionStorage.visited){
+      if (isOpera) {
+        alert(browserWarningMsg)
+      }
```
Comment (Weiqi97) [inline lexos/static/js/scripts_upload.js]: Use the Eslint forced style:

```javascript
if ( ) {
  // Code
} else if ( ){
  // Code
} ...
```

**Comment by chantisnake** [comment]
Don't write your own code use the npm package.

**Comment by chantisnake** [comment]
Browser detection will be hard to maintain. We have to use a package for this.

**Comment by mathewleblanc** [comment]
All of the packages I can find with npm use `require()` which is not supported in default Javascript but is supported in Node.js. If anyone can find a package which does not use require that would be great.

**Comment by chantisnake** [comment]
`require` is not only supported in node. You can just use import. Or you can just include them in HTML and use the function in the file.

**Comment by mathewleblanc** [comment]
I'm trying to use `import 'detect-browser'` but I keep getting this error:
`SyntaxError: import declarations may only appear at top level of a module`
I can't find much information on how to fix this. I think `import` is definitely the best way to do this but I do not know how to resolve this. [This](https://www.npmjs.com/package/detect-browser) is the package I am trying to use

**Comment by chantisnake** [comment]
I think the error message is pretty clear. You need to write import at the beginning of a file.

**Comment by mathewleblanc** [comment]
That is exactly what I did. It did not work.

**Comment by chantisnake** [comment]
What does Google tell you, my friend.

**Pull request closed without merging**